tfhe/src/core_crypto/commons/math/decomposition/decomposer.rs

```diff
@@ -353,40 +390,96 @@ where
     /// use tfhe::core_crypto::commons::parameters::{
     ///     CiphertextModulus, DecompositionBaseLog, DecompositionLevelCount,
     /// };
+    ///
+    /// let decomposition_base_log = DecompositionBaseLog(4);
+    /// let decomposition_level_count = DecompositionLevelCount(3);
+    /// let cihertext_modulus = CiphertextModulus::try_new((1 << 64) - (1 << 32) + 1).unwrap();
+    ///
     /// let decomposer = SignedDecomposerNonNative::new(
-    ///     DecompositionBaseLog(4),
-    ///     DecompositionLevelCount(3),
-    ///     CiphertextModulus::try_new((1 << 64) - (1 << 32) + 1).unwrap(),
+    ///     decomposition_base_log,
+    ///     decomposition_level_count,
+    ///     cihertext_modulus,
     /// );
     ///
+    /// let basis = 2i64.pow(decomposition_base_log.0.try_into().unwrap());
+    /// let half_basis = basis / 2;
+    ///
     /// // These two values allow to take each arm of the half basis check below
     /// for value in [1u64 << 63, 16982820785129133100u64] {
     ///     for term in decomposer.decompose(value) {
     ///         assert!(1 <= term.level().0);
     ///         assert!(term.level().0 <= 3);
-    ///         let term = term.value();
-    ///         let abs_term = if term < decomposer.ciphertext_modulus().get_custom_modulus() as u64 / 2
-    ///         {
-    ///             term
-    ///         } else {
-    ///             decomposer.ciphertext_modulus().get_custom_modulus() as u64 - term
-    ///         };
-    ///         println!("abs_term: {abs_term}");
-    ///         let half_basis = 2u64.pow(4) / 2u64;
-    ///         println!("half_basis: {half_basis}");
-    ///         assert!(abs_term <= half_basis);
+    ///         let signed_term = term.value().into_signed();
+    ///         assert!(-half_basis <= signed_term);
+    ///         assert!(signed_term < half_basis);
```

Comment:
See the above comment. With input  2^63 - 2^32 + 1 = 9223372032559808513 this second assert fails because of the strict inequality. This should be replaced by a non-strict inequality.